app/src/main/java/fr/free/nrw/commons/CommonsAppAdapter.java

```diff
@@ -7,15 +7,22 @@
 import org.wikipedia.login.LoginResult;
 
 import javax.inject.Inject;
+import javax.inject.Singleton;
 
 import androidx.annotation.NonNull;
 import fr.free.nrw.commons.auth.SessionManager;
 import okhttp3.OkHttpClient;
 
+@Singleton
 public class CommonsAppAdapter extends AppAdapter {
     private final int DEFAULT_THUMB_SIZE = 640;
 
-    @Inject SessionManager sessionManager;
```

Comment:
The app was crashing for me when I tried using `AppAdapter.get().getUserName`. So I tried fixing the injection. But from your comments it seems, I have gotten the library usage wrong. Will wait for you to complete your changes before trying it our further. :)